Apps/WebClient/src/ClientApp/src/components/timeline/entryCard/DiagnosticImagingTimelineComponent.vue

```diff
@@ -0,0 +1,164 @@
+import EntryCardTimelineComponent from "@/components/timeline/entryCard/EntrycardTimelineComponent.vue";
+import { EntryType, entryTypeMap } from "@/constants/entryType";
+import DiagnosticImagingTimelineEntry from "@/models/diagnosticImagingTimelineEntry";
+import { PatientDataFile } from "@/models/patientDataResponse";
+import container from "@/plugins/container";
+import { SERVICE_IDENTIFIER } from "@/plugins/inversify";
+import { ILogger } from "@/services/interfaces";
+import SnowPlow from "@/utility/snowPlow";
+
+const options = {
+    components: {
+        EntryCard: EntryCardTimelineComponent,
+        MessageModalComponent,
+    },
+};
+@Component(options)
+export default class DiagnosticImagingTimelineComponent extends Vue {
+    @Prop({ required: true })
+    hdid!: string;
+
+    @Prop()
+    entry!: DiagnosticImagingTimelineEntry;
+
+    @Prop()
+    index!: number;
+
+    @Prop()
+    datekey!: string;
+
+    @Prop()
+    isMobileDetails!: boolean;
+
+    @Prop({ default: false })
+    commentsAreEnabled!: boolean;
+
+    @Action("retrievePatientDataFile", { namespace: "patientData" })
+    retrievePatientDataFile!: (params: {
+        fileId: string;
+        hdid: string;
+    }) => Promise<PatientDataFile>;
+
+    @Getter("isPatientDataFileLoading", { namespace: "patientData" })
+    isPatientDataFileLoading!: (fileId: string) => boolean;
+
+    @Ref("messageModal")
+    readonly messageModal!: MessageModalComponent;
+
+    private logger!: ILogger;
+
+    get isLoadingFile(): boolean {
+        return (
+            this.entry.fileId !== undefined &&
+            this.isPatientDataFileLoading(this.entry.fileId)
+        );
+    }
+
+    get entryIcon(): string | undefined {
+        return entryTypeMap.get(EntryType.DiagnosticImaging)?.icon;
+    }
+
+    created(): void {
+        this.logger = container.get<ILogger>(SERVICE_IDENTIFIER.Logger);
+    }
+
+    showConfirmationModal(): void {
+        this.messageModal.showModal();
+    }
+
+    downloadFile(): void {
+        if (this.entry.fileId) {
+            SnowPlow.trackEvent({
+                action: "download_report",
+                text: "Diagnostic Imaging PDF",
+            });
+            const dateString = this.entry.date.format("yyyy_MM_dd-HH_mm");
+            this.retrievePatientDataFile({
+                fileId: this.entry.fileId,
+                hdid: this.hdid,
+            })
+                .then(
+                    (patientFile: PatientDataFile) =>
+                        new Blob([new Uint8Array(patientFile.content)], {
+                            type: patientFile.contentType,
+                        })
+                )
+                .then((blob) =>
+                    saveAs(blob, `diagnostic_image_${dateString}.pdf`)
+                )
+                .catch((err) => this.logger.error(err));
+        }
+    }
+}
+</script>
+
+<template>
+    <EntryCard
+        :card-id="`${index}-${entry.id}`"
```

Comment:
We usually use `datekey` instead of the `entry.id` here.  The combination of `datekey` and `index` should be unique.  Was there a reason to use the id instead?

Reply:
no I already went through the trouble to expose the id so I figured I would use it.